gold/testsuite/Makefile.in

```diff
@@ -8626,6 +8646,18 @@ uninstall-am:
 @DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@nanomips_move_balc_2_out_of_range_pcrel: nanomips_move_balc_out_of_range_pcrel.o ../ld-new
 @DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@	../ld-new --section-start .text=0x1000 --section-start .foo=0x2001020 -o $@ $<
 
+@DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@nanomips_long_balc_bc_offset_1: nanomips_long_balc_bc_offset.o ../ld-new
+@DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@	../ld-new --expand-long-balc-bc --section-start .text=0x1000 --section-start .foo=0x1fe1002 -o $@ $<
+
+@DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@nanomips_long_balc_bc_offset_2: nanomips_long_balc_bc_offset.o ../ld-new
+@DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@	../ld-new --section-start .text=0x1000 --section-start .foo=0x1fe1002 -o $@ $<
+
+@DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@nanomips_long_balc_bc_offset_3: nanomips_long_balc_bc_offset.o ../ld-new
+@DEFAULT_TARGET_NANOMIPS_TRUE@@NATIVE_OR_CROSS_LINKER_TRUE@	../ld-new --expand-long-balc-bc --section-start .text=0x1000 --section-start .foo=0x1fe1004 -o $@ $<
```

Comment:
Please add one more variation of tests 1 & 3 with positions of .text and .foo reversed to check the -ve range. Addresses will need some tweaking as offset is calculated backwards from NextPC.